boskos/ranch/storage.go

```diff
@@ -266,13 +262,21 @@ func (s *Storage) updateDynamicResources(lifecycle common.DynamicResourceLifeCyc
 	for _, r := range resources {
 		// We can only delete resources not in use
 		if !r.IsInUse() {
-			// Expired
-			if r.ExpirationDate != nil && s.now().After(*r.ExpirationDate) {
+			// deleting resources already marked for deletion, and not including them in the count.
+			if r.State == common.Tombstone || r.State == common.ToBeDeleted {
```

Comment:
hummmmmm... so as of today, the actual boskos server doesn't care about the actual state, and all state transitions are happening in controller's API calls... Can this be done by the cleaner? Aka cleaner can just delete the CRD instead.

Reply:
I would rather only have boskos delete things (because it has the lock on all resources), and have the other components just have read access.

Reply:
Or should this only affect for dynamic resources? (For a static resource, seems it will be deleted, and got recreated again in next cycle)

Reply:
currently I changed both for consistency.

Reply:
What do you think ?

Reply:
I think it's a no-op and static resources should not get into these states.. we can see how it works

Reply:
OK I can change it for static resources.